fugue/dataframe/pandas_dataframe.py

```diff
@@ -75,6 +76,8 @@ def __init__(  # noqa: C901
                 pdf, schema = self._apply_schema(pdf, schema)
             super().__init__(schema, metadata)
             self._native = pdf
+        except SchemaError as e:
```

Comment:
The problem is currently, the dataframe test suite expects to see `FugueDataFrameInitError`, so I was also thinking about this, I am not sure if the current behavior makes sense or not.

One quick thing we can do is change line 82 to `raise FugueDataFrameInitError(str(e))`, but again, I think we can chat about the better behavior for exceptions.